src/main/java/net/minecraftforge/common/extensions/IForgeResourceKey.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.Holder;
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+
+import java.util.Objects;
+
+public interface IForgeResourceKey<T> {
+    private ResourceKey<T> self() {
+        return (ResourceKey<T>) this;
+    }
+
+    default Holder<T> getOrThrow(BlockEntity blockEntity) {
+        return getOrThrow(blockEntity.getLevel());
+    }
+
+    default Holder<T> getOrThrow(Entity entity) {
+        return getOrThrow(entity.registryAccess());
+    }
```

Comment:
Same for this. The entity's registry access method is simply calling its level reference's. Just using the single `getOrThrow(Level)` method is probably for the best, so as to not cause confusion.

Reply:
He and I talked about this, these are meant to be helper functions that yes are strictly not necessary but people have complained that they want easier access.